Dockerfile

```diff
@@ -779,6 +781,63 @@ RUN apt-get update \
 
 FROM proteus_install_ptzendnn_${ENABLE_PTZENDNN} as proteus_dev_final
 
+
+# Install migraphx which supports GPU targets.  At the time of writing this,
+# it was necessary to build migraphx from source but it should be possible to install from repo
+# instead.
+#
+# Add rocm repository
+RUN sh -c 'echo deb [arch=amd64 trusted=yes] http://repo.radeon.com/rocm/apt/5.0/ ubuntu main > /etc/apt/sources.list.d/rocm.list'
+# Install dependencies for migraphx
+RUN apt-get update &&\
+    apt-get install -y sudo git bash build-essential rocm-dev libpython3.6-dev python3-dev python3-pip miopen-hip-dev \
```

Comment:
build-essential, git, sudo and some other packages are already present. You don't need to install them again. That also allows you to not need to reset gcc below. Once you combine these into one RUN command, you should also delete any packages that are only needed at build-time

Reply:
All except the gcc version reset, done. gcc version still needed for builds to work within the Docker container; not clear what's changing it.  Commit 765a081

Reply:
See commit 71b0aa1627.  The gcc version reset turns out to still be necessary.